UI/src/app/components/dialog/dialog.component.html

```diff
@@ -0,0 +1,84 @@
+<mat-card class="dialog">
+  <mat-list>
+    <ng-container *ngFor="let comment of comments">
+      <ng-template 
+      [ngTemplateOutlet]="loadTemplate(comment)" 
+      [ngTemplateOutletContext]="{ $implicit: comment}">
+      </ng-template>
+    </ng-container>
+  </mat-list>
+
+  <input [(ngModel)]="value" [disabled]="isDisabled()" placeholder="Input comment" class="input input-add" (keyup.enter)="add()">
+
+  <button mat-button (click)="add()" [disabled]="isDisabled()" class="button-add" disableRipple="true">
+    Add
+  </button>
+</mat-card>
+
+<ng-template #readOnlyTemplate let-comment>
+  <mat-list-item class="list-item-read">
+    <div class="comment-read">
+      <div class="comment-read-head">
+        <span class="comment-read-date">
+          {{comment.date | date:'MMM d'}}
+          <ng-container *ngIf="comment.hasBeenEdited">
+            <span>(edited)</span>
+          </ng-container>
+        </span>
+        <div class="comment-read-head-buttons">
```

Comment:
You have nesting, you use preprocessor SCSS. There is no sense to make so long names for css classes